app/task_routes.py

```diff
@@ -0,0 +1,131 @@
+from app import db
+from app.models.task import Task
+from flask import Blueprint, jsonify, abort, make_response, request
+import datetime
+import requests
+from app import token 
+
+
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+
+def validate_task(id):
+    try:
+        id = int(id)
+    except:
+        abort(make_response({"message": f"Task {id} is invalid"}, 400))
+
+    task = Task.query.get(id)
+
+    if not task:
+        abort(make_response({"message": f"Task {id} not found"}, 404))
+
+    return task
+
+@tasks_bp.route("", methods=["POST"])
+def create_task():
+    if request.method == "POST":
+        request_body = request.get_json()
+        if "title" not in request_body or "description" not in request_body:
+            return make_response(jsonify({"details": "Invalid data"}), 400)
+
+    new_task = Task(
+        title = request_body["title"],
+        description = request_body["description"],
+        completed_at = None
+    )
```

Comment:
Here is where we could use the from_dict method. So this part of the route would look like:
```python
new_task = Task.from_dict(request_body)
```

And lines 32-24 would be inside the method in the Task class, which would make this route more concise.